Meta/build-native-partition.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+
+set -e
+
+die() {
+    echo "die: $*"
+    exit 1
+}
+
+if [ "$(id -u)" != 0 ]; then
+    sudo -E -- "$0" "$@" || die "this script needs to run as root"
+    exit 0
+else
+    : "${SUDO_UID:=0}" "${SUDO_GID:=0}"
+fi
+
+
+if [ -z "$SERENITY_TARGET_INSTALL_PARTIION" ]; then
+    die "SERENITY_TARGET_INSTALL_PARTIION environment variable was not set!"
+fi
+
+printf "mounting filesystem... "
+mkdir -p mnt
+if ! eval "mount $SERENITY_TARGET_INSTALL_PARTIION mnt/"; then
+    die "Could not mount existing ext2 filesystem on $SERENITY_TARGET_INSTALL_PARTIION"
+else
+    echo "done"
+fi
+
+cleanup() {
+    if [ -d mnt ]; then
+        umount mnt || ( sleep 1 && sync && umount mnt )
+        rmdir mnt
+        echo "done"
+    fi
+}
+trap cleanup EXIT
```

Comment:
Shouldn't we set up the trap before the `mkdir`?

Reply:
YUP!